api/src/auth/drivers/oauth2.ts

```diff
@@ -163,7 +153,7 @@ export class OAuth2AuthDriver extends LocalAuthDriver {
 				: generators.codeChallenge(payload['codeVerifier']);
 
 			tokenSet = await this.client.oauthCallback(
-				this.redirectUrl,
+				payload['redirectUri'],
```

Comment:
We name it `callbackUrl` when we pass it into `authenticationService.login` on L453, this will result in `redirectUri` being undefined